sopel/modules/translate.py

```diff
@@ -158,7 +159,16 @@ def tr2(bot, trigger):
         return
 
     def langcode(p):
-        return p.startswith(':') and (2 < len(p) < 10) and p[1:].isalpha()
+        # TODO: We'd be much better off just using the langcodes PyPI package
+        # also TODO: it'd be nice not to require the : prefix, which using
+        # langcodes would make easier in lieu of adding an API key to fetch
+        # the list of supported languages
+        prefixed = p.startswith(':')
+        # the longest codes Google uses (ca. Jan 2022) are zh-CN and zh-TW
+        short_enough = (2 < len(p) < 8)
+        # pesky - in Chinese codes forced a switch from .isalpha(); see #2241
+        fits_pattern = re.match(r':[a-z\-]+', p.lower())
```

Comment:
```suggestion
        fits_pattern = re.match(r':[a-z\-]+', p, re.IGNORECASE)
```

Reply:
I almost did this, tbh. Then I decided it wasn't worth the effort of amending the commit (and making GHA re-test).

Reply:
Did you try to compile the regex first? 😁

Reply:
I thought it was memoized, but maybe `timeit` skips over that built-in optimization.

```pycon
>>> timeit('r.match(s.lower())', 'import re; s = ":zh-CN"; r = re.compile(r":[a-z\-]+")')
0.19142690021544695
>>> timeit('r.match(s)', 'import re; s = ":zh-CN"; r = re.compile(r":[a-z\-]+", re.IGNORECASE)')
0.2070107003673911
```

So it's _really_ a "who cares?" moment, then. 😂